test/e2e/specs/interactivity/directive-each.spec.ts

```diff
@@ -500,4 +500,37 @@ test.describe( 'data-wp-each', () => {
 		await expect( element ).toHaveText( 'beta' );
 		await expect( callbackRunCount ).toHaveText( '1' );
 	} );
+
+	for ( const testId of [
+		'each-with-unset',
+		'each-with-null',
+		'each-with-undefined',
+	] ) {
+		test( `does not error with non-iterable values: ${ testId }`, async ( {
+			page,
+		} ) => {
+			await expect( page.getByTestId( testId ) ).toBeEmpty();
+		} );
+	}
+
+	for ( const [ testId, values ] of [
+		[ 'each-with-array', [ 'an', 'array' ] ],
+		[ 'each-with-set', [ 'a', 'set' ] ],
+		[ 'each-with-string', [ 's', 't', 'r' ] ],
+		[ 'each-with-generator', [ 'a', 'generator' ] ],
+
+		// TODO: Is there a problem with proxies here?
+		// [ 'each-with-iterator', [ 'implements', 'iterator' ] ],
```

Comment:
Oh, I realized you meant there's a problem only for the `each-with-iterator` case which is commented out.

Reply:
That's right, it's comments out because it would not pass.

I didn't spend much time debugging the problem so I'm not sure what the issue is. It's some thing that apparently should work but does not, I'm not sure of the cause and don't really plan to spend time debugging it, maybe @DAreRodz know if this would create issues with store proxies.